### DISCUSSION TITLE
I am so lost when it comes to custom components. Please Help

### DISCUSSION THREAD

**Discussion by properCase** [vuepress-theme-hope/vuepress-theme-hope#2178] [Q&A] [answered]
Ok, the documentation as I read it states that as long as you register your Vue Components you can access them from your markdown files.  However the documentation is super sparing on details of _where_ we are supposed to register them and how.

Here is my tree structure:
```
docs/
    .vuepress/
        components/
            HelloVue.vue // The component
        enhanceApp.js  // <-- the place I think I'm supposed to register them
```

docs/.vuepress/components/HelloVue.vue
```html
<template>
  <h1>HELLO VUE!</h1>
</template>

<script>
export default {
  name: 'HelloVue'
}
</script>

<style scoped>

</style>
```

The below is supposed to load whatever components are in the folder:

docs/.vuepress/enhanceApp.js
```js
// @ts-ignore
const modules = import.meta.globEager('../components/**/*.vue')
const components = []

for (const path in modules) {
  components.push(modules[path].default)
}

export default ({
  Vue, // the version of Vue being used in the VuePress app
  options, // the options for the root Vue instance
  router, // the router instance for the app
  siteData // site metadata
}) => {
  components.forEach(component => {
    Vue.component(component.name, component)
  })
}

```

docs/test/README.md
```md
# This is a test of the component

Text Before

<HelloVue />

Text After
```

The page renders just the text.  There is an html tag: `<hellovue></hellovue>` but nothing else.

Clearly I'm doing something wrong.

**Comment by Mister-Hope** [comment, marked as answer]
Nope. try not using file based apis. Also where does "Vue" comes from? I think you are not using V1 right? if so, change the api below to V1, also v1 does not get any support (See announcement)

If you miss the docs, I would be very disappointed because of time waste: https://vuepress-theme-hope.github.io/v2/cookbook/advanced/component.html

```ts
// .vuepress/config.ts
import { defineClientConfig } from "@vuepress/client";
import Comp from "./components/Comp.vue";
import Comp2 from "./components/Comp2.ts";
import Comp3 from "./components/Comp3.js";

export default defineClientConfig({
  enhance: (app) => {
    app.component('Comp', Comp);
    app.component('Comp2', Comp2);
    app.component('Comp3', Comp3);
  },
});
```

Or use `@vuepress/plugin-register-components`.

**Reply by properCase** [reply]
Thank you for the reply, I am definitely on v2.

I think what is confusing me, (and I apologize, I am new to VuePress and your theme) is using the config.ts which had `defineUserConfig` being returned as default.

Now that I'm reading this I was able to go back and it seems that the above config should go in the client.ts, and I leave the config.ts as I had it.

I've done that and now the components are loading!
